octodns/provider/route53.py

```diff
@@ -1253,7 +1253,16 @@ def _mod_Delete(self, change, zone_id, existing_rrsets):
         return self._gen_mods('DELETE', existing_records, existing_rrsets)
 
     def _extra_changes_update_needed(self, record, rrset):
-        healthcheck_host = record.healthcheck_host
+        value = rrset['ResourceRecords'][0]['Value']
+
+        try:
+            ip_address(text_type(value))
+            # We're working with an IP
+            healthcheck_host = record.healthcheck_host
+        except (AddressValueError, ValueError):
+            # This isn't an IP, host is the value
+            healthcheck_host = value
```

Comment:
Would it be more direct to check `record._type` here and return `value` if it's `CNAME` and `record.healthcheck_host` otherwise? I believe this would work, but it took me staring at it and thinking about the cases for a couple minutes before I thought so. The if type route might be a little more explicit and immediately clear (unless I'm missing something.)

Reply:
Yes, that should work too and I agree it would be more explicit.  I'll prepare an update to this PR with the changes.  Thanks for reviewing.